src/libcore/slice/mod.rs

```diff
@@ -2998,14 +2998,27 @@ macro_rules! is_empty {
 // unexpected way. (Tested by `codegen/slice-position-bounds-check`.)
 macro_rules! len {
     ($self: ident) => {{
+        #![allow(unused_unsafe)] // we're sometimes used within an unsafe block
+
         let start = $self.ptr;
-        let diff = ($self.end as usize).wrapping_sub(start as usize);
         let size = size_from_ptr(start);
         if size == 0 {
+            // This _cannot_ use `unchecked_sub` because we depend on wrapping
+            // to represent the length of long ZST slice iterators.
+            let diff = ($self.end as usize).wrapping_sub(start as usize);
             diff
```

Comment:
There is now no need for `diff` here

Reply:
@vitiral I'm quite confident it's needed, but do feel free to make a PR demonstrating otherwise.

Reply:
I think they mean that the `let` here is redundant, and I agree. `let diff = e; diff` is the same as `e`.